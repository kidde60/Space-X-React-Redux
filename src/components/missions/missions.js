import React, { useEffect } from 'react';
import { createAsyncThunk } from "@reduxjs/toolkit";
import { useDispatch, useSelector } from 'react-redux';
import axios from 'axios';
import MissionCard from './missionCard';
import { getMissions } from '../../redux/Mission/mission'

const Mission = () => {
  const Missions = useSelector((state) => state.missions);
  const dispatch = useDispatch

  const fetchMisssions = createAsyncThunk(
    "missions/fetch/missions",
    async () => {
      const missions = await fetch("https://api.spacexdata.com/v3/rockets");
      const data = await missions.json();
      console.log(data)
      return data;
    },
  );
  fetchMisssions()

  // const missionUrl = 'https://api.spacexdata.com/v3/missions'
  // const fetchMission = async () => {
  //   const response = await axios.get(missionUrl).catch((error) => {
  //     console.log('error', error)
  //   })
  //   console.log(response.data)
  //   dispatch(getMissions(response.data))

  // }
  // useEffect(() => {
  //   fetchMission();
  // }, []);
  // console.log(Missions);
  // return (
  //   <MissionCard />

  // );
};

export default Mission;