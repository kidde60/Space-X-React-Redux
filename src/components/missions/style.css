.listItems {
  border: 1px solid gray;
  box-shadow: 1px 1px 10px 3px;
  margin: 20px 50px;
}

.listItem {
  border: 1px solid gray;
  margin: 20px 50px;
}
