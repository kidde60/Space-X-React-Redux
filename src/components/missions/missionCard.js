import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Missions from './missions'
import { joinMission } from '../../redux/Mission/mission'
import './style.css';

const MissionCard = () => {
    const missionList = useSelector((state) => state.missions)
    const dispatch = useDispatch();
    const join = (e) => {
        e.preventDefault()
        dispatch(joinMission(e.target.id));
    }

    return (
        <section >
            <ul className="list">
                <li className="listItem">
                    <h2 className="heading">Mission</h2>
                    <h2 className="paragraph">Description</h2>
                    <h2>Status</h2>
                </li>
                {
                    missionList.map((mission) => {
                        const {
                            mission_id, mission_name, description, reserved,
                        } = mission;
                        return (
                            <Missions
                                key={mission_id}
                                id={mission_id}
                                name={mission_name}
                                desc={description}
                                join={join}
                                isReserved={reserved}
                            />
                        );
                    })
                }
            </ul>
        </section>
    );
};

export default MissionCard;
