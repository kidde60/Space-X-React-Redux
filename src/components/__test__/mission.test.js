import renderer from 'react-test-renderer';

import missions from '../missions/missions';

const mission = {
    name: 'SpaceX',
    desc: 'Lorem ipsum text generate',
    id: '56DFRE',
    join: () => true,
    reserved: false,
};

const mission2 = {
    name: 'SpaceX',
    desc: 'Lorem ipsum text generate',
    id: '56DFRE',
    join: () => true,
    reserved: true,
};

it('<missions /> renders properly as expected', () => {
    const {
        name, desc, id, join, reserved,
    } = mission;
    const tree = renderer.create(<missions
        name={name}
        id={id}
        desc={desc}
        join={join}
        reserved={reserved}
    />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('<missions /> renders properly as expected', () => {
    const {
        name, desc, id, join, reserved,
    } = mission2;
    const tree = renderer.create(<missions
        name={name}
        id={id}
        desc={desc}
        join={join}
        reserved={reserved}
    />).toJSON();
    expect(tree).toMatchSnapshot();
});