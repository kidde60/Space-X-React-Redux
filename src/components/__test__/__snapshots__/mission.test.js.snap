// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<missions /> renders properly as expected 1`] = `
<missions
  desc="Lorem ipsum text generate"
  id="56DFRE"
  join={[Function]}
  name="SpaceX"
  reserved={false}
/>
`;

exports[`<missions /> renders properly as expected 2`] = `
<missions
  desc="Lorem ipsum text generate"
  id="56DFRE"
  join={[Function]}
  name="SpaceX"
  reserved={true}
/>
`;
